homeassistant/components/ozw/light.py

```diff
@@ -133,3 +189,42 @@ async def async_turn_off(self, **kwargs):
         self.async_set_duration(**kwargs)
 
         self.values.primary.send_value(0)
+
+    def _calculate_rgb_values(self):
+        # Color Channels
+        self._color_channels = self.values.color_channels.data[ATTR_VALUE]
+
+        # Color Data String
+        data = self.values.color.data[ATTR_VALUE]
+
+        # RGB is always present in the openzwave color data string.
+        rgb = [int(data[1:3], 16), int(data[3:5], 16), int(data[5:7], 16)]
+        self._hs = color_util.color_RGB_to_hs(*rgb)
+
+        # Parse remaining color channels. Openzwave appends white channels
+        # that are present.
+        index = 7
+
+        # Warm white
+        if self._color_channels & COLOR_CHANNEL_WARM_WHITE:
+            warm_white = int(data[index : index + 2], 16)
+            self._white = warm_white
+            index += 2
```

Comment:
Why do we increase the index here? Are the warm and cold cases mutually exclusive?

Reply:
> Why do we increase the index here? Are the warm and cold cases mutually exclusive?
They are WW comes before CW

Reply:
So the cases are not mutually exclusive? If the light is CW will it then always be WW too?

Reply:
No there can be CW w/o WW and vice versa.

Reply:
So if a light is only cold white, it looks like we're modifying the same index interval as for the warm white currently. That doesn't look correct.

Reply:
It increments it along the #RRGGBBWWCW value. We're starting at 7 which is #RRGGBB|WWCW then the CW value, #RRGGBBWW|CW

Reply:
Yes, so if the light is CW only, the index will still be 7 currently. That doesn't seem correct.

Reply:
Ah I see what you mean, sorry I'm not trying to be dense lol

Reply:
Fixed in latest commit.